### DISCUSSION TITLE
Milton fried my Pi SD card!

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#1035] [App and O/S Config Questions]
So Milton happened... despite being in the eye of the storm, somehow we never lost power and my pool solar panels stayed on the roof!

The power did flicker tho causing the Pi to shutdown and I think the SD card has become corrupted.  Services are still running but I noticed REM config was wiped and I've had to restore the `data/controllerConfig.json` for 3 days straight now.  

2 questions:

1. If I create another SD card and install the njspc stack, what's the best way to restore?  I have a year of daily backups - just copy the json files from a recent backup to their location in the app folders?

2. How about preventing corruption in the future?  Do you guys run your SD in RO mode?  

I had tried sequent's watchdog battery backup in the past but it fried on me and I never got around to replacing it.

**Comment by FlaMike** [comment]
I posted  a response the day you posted your tale of woe above.  I have no idea why it never appeared in the thread.  I'll try to recreate it, but I don't have a copy so it will certainly differ from my thoughts back when...

First and  foremost, so glad that Milton didn't impact you more than lights flickering and a fried SD card.  It's been a rough fall on the Gulf Coast!  We're far enough west in the  Panhandle that we were unaffected by both storms.  Knock on wood & fingers crossed for everyone going forward!!!

For what it's  worth, I maintain an extra SD card that's a clone of the one running in the Pi because , well, they're SD cards!  Not the most stable storage on the planet, for certain.  I've also cloned the system onto an SSD and will likely switch to it someday in the hope it's a bit more robust.  I have an enclosure for it that runs off USB.  So, it's plug and play when it needs to go into service.

I don't have an answer to your first question--I just cloned a working SD card verbatim using Balena Etcher & keep it in the house.  The biggest difference over time may be a few setpoints that I play with occasionally and the logs.  Neither concerns me if I have  to swap cards.

Per your second question, I do not run the SD in RO mode.  Others may vary, of course.

I hope you & your Nixie have fully recovered from Milton!

**Reply by johnny2678** [reply]
thanks for responding!

I've been running around like crazy since posting this getting repairs in line while travelling for work. I finally had some time to tackle this yesterday.  In the end, a simple backup of the corrup SD card and restore to a clean one ended up working.  I was surprised, but the new SD card is writable and doesn't reset/revert to a prior state every 24 hours.  So I'm back in business!

**Reply by FlaMike** [reply]
Glad to hear your SD card woes are history.  I hope all other aspects of Milton's impact to you and family, friends, & others in its path and  nearby are history or will be soon.  What a terrible month it's been.  Neighbors are hosting friends who lived in Asheville & were in Helene's wake.  Fortunately, they're fine & their home was intact when they exited a week after the storm.  Of course,  the impact to NC was massive and will take years to recover.

Here's to happier days ahead for all!
